openupgrade_scripts/scripts/hr_expense/16.0.2.0/post-migration.py

```diff
@@ -0,0 +1,66 @@
+# Copyright 2023 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from openupgradelib import openupgrade
+
+
+def _hr_expense_process(env):
+    """Set amount_tax_company value only when it is necessary with ORM."""
+    env.cr.execute(
+        """
+        SELECT id
+        FROM hr_expense
+        WHERE total_amount_company != total_amount AND amount_tax > 0
+        AND amount_tax_company != 0
+        """
+    )
+    expense_model = env["hr.expense"]
+    for expense_id in env.cr.fetchall():
+        expense = expense_model.browse(expense_id)
+        expense._compute_amount_tax()
```

Comment:
In general, @pedrobaeza recommends to not call compute function, but rather implement it in SQL. The risk by calling a compute method is that it generate other compute that can give undesired side effects. Am I right ?

Reply:
Yes, and for performance reasons, but sometimes there's no equivalent SQL or the number of records to compute is very low that it doesn't worth. Didn't check if this is the case.